client/src/containers/NotRegisteredPage.tsx

```diff
@@ -39,20 +54,19 @@ export default class NotRegisteredPage extends Component {
         ? this.state.buildingInfo[0]
         : null;
 
-    const usersInputAddress =
-      searchAddress && searchAddress.boro && (searchAddress.housenumber || searchAddress.streetname)
-        ? {
-            boro: searchAddress.boro,
-            housenumber: searchAddress.housenumber || " ",
-            streetname: searchAddress.streetname || " ",
-          }
-        : buildingInfo && buildingInfo.boro && (buildingInfo.housenumber || buildingInfo.streetname)
-        ? {
-            boro: buildingInfo.boro,
-            housenumber: buildingInfo.housenumber || " ",
-            streetname: buildingInfo.streetname || " ",
-          }
-        : null;
+    const usersInputAddress = searchAddress
+      ? {
+          boro: searchAddress.boro,
+          housenumber: searchAddress.housenumber || " ",
+          streetname: searchAddress.streetname || " ",
+        }
+      : buildingInfo
+      ? {
+          boro: buildingInfo.boro,
+          housenumber: buildingInfo.housenumber || " ",
+          streetname: buildingInfo.streetname || " ",
+        }
+      : null;
```

Comment:
I know this PR is just for typescript-ification but it seems like we could just factor out this object notation into a helper function to make things more DRY.  If I'm correct maybe we should make an issue for it or something?  Not really a big deal either way I guess.

Reply:
Ahhh so actually there is an issue here—  `searchAddress.streetname` is typed to never be null so there isn't exact parity between the construction of the first object and the construction of the second one here.

Reply:
I just updated the code to reflect that.